### PR TITLE
build: automate changelog

### LINKED ISSUE #19645: Feature Request: Automate Changelogging
### Background
Currently, our process for change logging is to create a PR, consider if the change belongs in changelog, add an entry, then force push the whole thing, referencing the PR.

Until late September, we didn't have a changelog in ui at all. It's an important concept for product and product marketing, but it's something we've let slide as engineers.

### Considerations
For a changelog to reference the PR it's part of, the PR has to be created before it can be linked in the changelog.

The relevant entries to include are `feat` and `fix`.

Should this be fully automated, or a script someone has to run?

### PR DESCRIPTION
**Pull request by codyshepherd**
Closes #19645 for `master-1.x`

Automates changelog maintenance as part of daily job. Changelog is moved to a public artifact in s3.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Review by davidby-influx — APPROVED** [review]
LGTM.  Would like another set of eyes on it, though.

File: .circleci/config.yml
Comment (wbaker85) [inline .circleci/config.yml:9]: Are there any options for downloading a pre-built `git-cliff` so we don't have to install rust here and build it locally?

Comment (codyshepherd) [reply .circleci/config.yml:9]: The GitHub project publishes tarballs with shasums and detached signatures, though it's a bit unclear to me at the moment where the public signing key is located.

Additionally, they public a Docker image via GitHub's container registry, but it's unclear to me at the moment how to verify the contents of this image in similar fashion to the tarball.

I agree that the rust build can be improved, but I'm not sure it's worth blocking this PR for it, since a) this logic mirrors what we have going in the other branches, aiding in maintainability, and b) the `changelog` job happens in parallel with other tests and such, and the hold-up it produces should be minimal.

My feeling would be that using some kind of pre-built package should be left for future work, since using Git is both correct and secure.

Comment (wbaker85) [reply .circleci/config.yml:9]: Makes sense to me!

**Review by wbaker85 — APPROVED** [review]

File: .circleci/config.yml
Comment (codyshepherd) [reply .circleci/config.yml:9]: I'll put up an issue for this. Thanks for the +1!

**Pull request merged**
